src/librustc/util/ppaux.rs

```diff
@@ -680,12 +680,16 @@ impl<'tcx> UserString<'tcx> for TraitAndProjections<'tcx> {
     fn user_string(&self, tcx: &ctxt<'tcx>) -> String {
         let &(ref trait_ref, ref projection_bounds) = self;
         let base = ty::item_path_str(tcx, trait_ref.def_id);
-        parameterized(tcx,
-                      &base,
-                      trait_ref.substs,
-                      trait_ref.def_id,
-                      &projection_bounds[..],
-                      || ty::lookup_trait_def(tcx, trait_ref.def_id).generics.clone())
+        if let Some(trait_def) = try_lookup_trait_def(tcx, trait_ref.def_id) {
+            parameterized(tcx,
+                          &base,
+                          trait_ref.substs,
+                          trait_ref.def_id,
+                          &projection_bounds[..],
+                          || trait_def.generics.clone())
+        } else {
+            format!("local_trait_not_ready({:?})", trait_ref.def_id)
```

Comment:
The `user_string` trait is supposed to eschew printing compiler internal details... I know this is an rare error conditon, so it might be acceptable here... but isn't the set of cases where this can arise limited to a small group (currently just `Send`)?  I wonder if we can leverage that somehow